.ado/jobs/e2e-test.yml

```diff
@@ -62,34 +62,12 @@ jobs:
           script: yarn windows --no-launch --no-packager --arch $(BuildPlatform) --release --logging --buildLogDirectory $(BuildLogDirectory) --msbuildprops BaseIntDir=$(BaseIntDir)
           workingDirectory: packages/E2ETest
 
-      - task: CopyFiles@2
-        displayName: Copy ReactUWPTestApp artifacts
-        inputs:
-          sourceFolder: $(Build.SourcesDirectory)/packages/E2ETest/windows/ReactUWPTestApp
-          targetFolder: $(Build.StagingDirectory)/ReactUWPTestApp
-          contents: AppPackages\**
-        condition: succeededOrFailed()
-
-      - task: PublishPipelineArtifact@1
-        displayName: "Publish Artifact:ReactUWPTestApp"
-        inputs:
-          artifactName: ReactUWPTestApp-$(System.JobAttempt)
-          targetPath: $(Build.StagingDirectory)/ReactUWPTestApp
-        condition: succeededOrFailed()
-
```

Comment:
Moved these to the end with other tasks only performed on failure, so we can avoid the steps of uploading dumps if not needed.